odbc/src/api/data.rs

```diff
@@ -584,12 +605,13 @@ pub unsafe fn format_datetime(
     match dt {
         Ok((dt, warning)) => {
             let data = Timestamp {
-                year: dt.year() as i16,
-                month: dt.month() as u16,
-                day: dt.day() as u16,
-                hour: dt.hour() as u16,
-                minute: dt.minute() as u16,
-                second: dt.second() as u16,
+                // should year be i16?
+                year: dt.year().try_into().expect("year exceeded i16 space"),
```

Comment:
Yes :)
You can find this in the struct definition : https://learn.microsoft.com/en-us/sql/odbc/reference/appendixes/c-data-types